### ISSUE TITLE
Improvement: private channels

### ISSUE THREAD

**Issue by infosquatter** [globaleaks/GlobaLeaks#1422]
It was asked by Peruleaks media organizations.

Need: to be able to communicate privately with the source AFTER submission is done. It applied both to messages and new uploaded files.

Problem: the claim is that in the case of a multiple receiver submission, one of the media organizations can start doing its investigation in alliance with the source and - as in the current situation - doing so could just be helping other organizations in their research, thus establishing an unfair competitive dynamic. A receiver should be allowed to communicate and receive files of a submission without helping other receivers in their investigations.

Worst case scenario: 

1. Source sends files to organization A and B. Organization A is not interested - at first - in the material (or do not have time to investigate it). Meanwhile, organization B is interested and have potential to investigate.
2. Organization B starts a conversation with the source because the files came incomplete or lacking means of authentication.
3. Source sends new files and give further relevant information for the investigation.
4. Step 2 and 3 repeat.
5. Organization A came back to the submission interface and see a lot of new stuff: more files, more information from the source. Organization A decided to publish the information very quickly.
6. Organization A publishes the information before organization B and organization B - who was really keen on making a deep investigation on the issue - loses exclusivity.

There a lot of other case scenarios which these conditions would be unfair. The basic idea is that an organization cannot take profit of the efforts of another organization in this fashion.

Also, source can have different degrees of trust regarding different organization, despite the fact it submitted the files to these different organizations.

Etc, etc.

My view: i totally agree. Also because I´m somehow confident that the messaging system of Globaleaks has never actually worked in the way we all expected (maybe because of this issue) and it seems the partners in Peru are really interested in this feature (probably due to conditions of whistleblowing in Peru, which they are all already used to).

**Comment by evilaliv3** [comment]
nice request @infosquatter 

let me clarify what we avoided so far and what i find doable:
- we avoided to allow the whistleblower to remove what he has sent given the fact that the fear is a human factor that will point the whistleblower to remove the data a lot of times after sending it
- i'm not against letting the whistleblower to reduce the selection of audience for new material, and this is something that is already done by the the messages feature for private messages.
- i'm now thinking if the process should appear hidden completely or all receivers should at least see that the comunication is happening even just to trigger competition. it's also possible that more competition may happen if you knoow that all is hidden.

@marcel: can you discuss this also in publeaks? i remember various suggestions by sybren in relation to minor changes that would engage competition between medias and so that would rise the quality of their work and of the process in general.

\cc @fpietrosanti @vecna

**Comment by marceloomens** [comment]
Yes, I will certainly bring this up in discussion with the Publeaks participants. A possible work-around for now would be that whistleblowers start a new tip by going through the submission process and sending new documents to a sub selection of earlier receivers.

**Comment by vecna** [comment]
looks good, but I think the more honest and simple way, will be that organisation "A" just press the button "I don't care about - I'm not gonna to follow it", because this kind of feedback in to the Tips will be added soon or later.

Pressing that button, exclude "A" from Tip updates.

I'm also ok with the selective update proposed, just I'm worried that would complex much more UX and simplicity of the usage (practically speaking, any content in the Tip page has to report which has been the recipient in that time)

And, additionally, I always heard about media that are afraid of publish not verified material, thinking that "A" publish out of the blue,... sounds like a journalist-paranoia-OMG-the-others-are-crazy-they-are-going-to-stole-our-scoops :) ?

**Comment by infosquatter** [comment]
@vecna . The "I dont care about" button won't work because relation between media is simply competitive. As long as they feel they can get extra info, they wont click thuis button.

IMPORTANT: In a final meeting yesterday with Peruleaks media, it was considered by local partners as an **essential** feature. The launch, that was set to 15oct, was postponed until this feature is implemented. 

I would like to know a) If you can/want implement it and b) when can it be implemented.

Implementing this feature is essential for Peruleaks launch, since partners prefer to postpone launching until it is implemented.

I dont think it is up to us to judge or criticize jorunalists' paranoia or histeria, but we should stick for the features they need and for their general petitions.

cc @evilaliv3 @fpietrosanti

**Comment by vecna** [comment]
Seems that users with just an **expectation** (not a proved experience), are requesting a feature. Normally this process works in a completely different way:

  * users use the platform, if they experience problem in managing the flow, they describe the problem.
  * users raise needs, and we figure out IF a technical feature is necessary, what has to be guarantee, how implement it.

This idea of having selective sharing, can be done, in theory (now see what imply). 

The UX in the side of the whistleblower (I'm afraid) will be quite complicated (and whistleblower is a sensitive point where complexity has always to be reduced). Because means that Tip file upload has to have an option **select recipients** where you can select a sub group. In the visualization has to be kept in account which are the recipients selected.

This feature require, at least:

  * the button "upload new evidences" will display a selection box of receivers, and this has to be nice with 1 or with > 20 (*1 day + new texts to be put in translations*)
  * the upload mechanism has to be internally changed, because now every file is associate to a Tip (and thus, to all the receivers) instead now, we've to figure out how associate the new files to a subset (*can't make a time estimation*)
  * a new security condition, if A (no PGP) and B and C (both PGP) are selected, the files are not encrypted. if B and C are selected for the second file, will the file be encrypted ? yes. but this will left our system with more conditions and the receiver with two different behavior to adopt. 
  * what if A delete the tip ? can't do "delete" in the moment the selective upload has been done ? (so is either or something that has to be communicated clearly, or a new condition)

@infosquatter Is not an issue dealing with messy feature. **is an issue, dealing with messy feature, decided by unexperienced receivers that make hypothesis of failure they have not yet experienced.**

@evilaliv3 is better "clone" the Tip for a reduced subset of users. imply two receipts or more, but at least don't create mess with new conditions. still, require a certain degree of UX and backend devs

**Comment by fpietrosanti** [comment]
This feature could be maybe implemented trying to address only minimal changes such as:

- Add on API an optional recipient identifier for a specific file upload
- Add on Database an optional recipient identifier for a specific file upload
- Change backend to return the file list for "those who have no recipient identifier" + "those who have recipienti identifier = logged receiver" 

- On the UX two modifications side:
  - while clicking "Upload file" rather than immediately opening up the file browser to select the file to be uploaded, provide a selector to choose the recipienti "All | Receiver1 | Receiver2 | Receiver3 | etc..." 
 - while displaying the file list, if a file has a specific recipient, display which is the receiver to which it has been sent

With this approach, it could be likely possible to implement the feature with minor code changes.

**Comment by vecna** [comment]
is unnecessary complicated @fpietrosanti. your assessment of "minor code changes" is not so simple how does it sounds. 

we can extend the "private message" feature with a "private file sharing", create a new DB called PersonalFiles and manage them separately.  This can be reasonable without touching all the stable sections of GL.

**Comment by fpietrosanti** [comment]
@vecna It's a nice idea to have the "action" to share a file to be triggered from a "private message" window.
But probably you will always need to visualize the files in the standard "File list", otherwise you will need to make additional custom UX elements to visualize it somewhere else in some other way, maybe missing additional features such as "Download all files", and would maybe need to have a different UX elements to show a progress bar for upload?

**Comment by infosquatter** [comment]
@vecna I understand this is a messy feature used by journalists that dont have experience **with the software**. But, as a matter of fact, they are indeed journalists used and with much experience with **sources**, and the software is just a tool for them to interact with their sources. I believe we should pay attention to their claims.

Are you working on that currently? If not, temporarily we are going to 1 receptor max. configurartion for Peru.

**Comment by evilaliv3** [comment]
no @infosquatter we are currently not working due to other hard schedule on the roadmap.

in addition all the changes that you are proposing are valuable but as they impact a lot on usability and security (we won't to add too much confusions to users on how the node can be configured; kiss! do one thing well is our motto!) we will need more time to discuss about this.

so my suggestion is that you try to achieve what you need with what is the release and never rely on an hard fast rush implementation cause this as now won't always easily doable.

**Comment by fpietrosanti** [comment]
This maybe related, for implementation purpose similarity (pieces of codes to be modified), also to #1191 and #1404
